docs/developer/HootenannyTests.asciidoc

```diff
@@ -0,0 +1,507 @@
+
+Hootenanny contains three distinct codebases: the core, the web services, and the user interface.  
+Each employs a different testing technology, which is described here.
+
+=== Test Suites
+
+Tests are divided into the following groups: current, quick, slow, and glacial.
+
+* current - Used when you want to isolate a single test and work on it. No tests should be committed 
+in this directory.
+* quick - Individual tests that each run in less than 1 sec.
+* slow - Individual tests that each run in less than 30 sec.
+* glacial - Individual tests that each run in less than 15 min.  You shouldn't need tests that take
+ longer than 15min.
+ 
+ If you want to run the tests for all codebases at the same time, you can use the following commands:
+
+-------------------
+./configure && make -sj($nproc) test-quick
+./configure && make -sj($nproc) test       # slow
+./configure && make -sj($nproc) test-all   # glacial
+------------------
+
+quick is a subset of the slow suite, and quick and slow are subsets of the glacial suite.
+
+As a developer, you are expected to, at a minimum, run the slow tests before pushing commits.  
+If you are doing user interface or web service work, you may also want to test your changes against 
+the User Interface Tests, which run as part of the glacial tests.
+
+=== Core Tests
+
+The Core Tests test everything that is accessible via the Hootenanny command line interface.  
+The core code is written in C++ and employs several different types of tests, which are described 
+in the following sections.  The tests are also subdivided into different test suites with differing 
+runtimes.
+
+To run all the Core Tests in a suite, choose one of the following:
+
+--------------------
+HootTest --current
+HootTest --quick
+HootTest --slow
+HootTest --glacial
+-------------------
+
+Here are some examples of more granularity when running Core Tests:
+
+---------------
+# Runs the glacial tests but excludes any test with 'Big' in the test file name.
+HootTest --glacial '--exclude=.*Big.*'
+
+# Runs the glacial tests but only includes tests with 'Big' in the test file name.
+HootTest --glacial '--include=.*Big.*'
+
+# Runs a single test
+HootTest --single hoot::AlphaShapeTest
+--------------
+
+==== Unit Tests
+
+The core unit tests use CPPUnit.  The goal is to have one test class for each functional class.  
+See hoot-core-test/src/test/cpp project for examples.  One CPPUnit test class may contain multiple 
+CPPUnit tests.
+
+To place a test into one of the four test suites listed in the previous section, a line similar to 
+the following is placed at the end of the test class:
+
+-----------------
+CPPUNIT_TEST_SUITE_NAMED_REGISTRATION(AlphaShapeTest, "quick");
+------------------ 
+
+==== Core Command Line Tests
+
+The C++ portion of Hootenanny exposes functionality via a command line interface. This command line 
+interface is tested via simple shell scripts similar to the Signature Analyst way of testing. The 
```

Comment:
In the context of redmine a reference to Signature Analyst was probably fine, but here it should probably be removed.